src/main/java/com/programmers/calculator/Accumulator.java

```diff
@@ -0,0 +1,52 @@
+package com.programmers.calculator;
+
+import com.programmers.converter.ExpressionConverter;
+import com.programmers.converter.Operator;
+import com.programmers.converter.ResultConverter;
+import com.programmers.io.Console;
+
+import java.math.BigDecimal;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.List;
+import java.util.Optional;
+
+public class Accumulator {
+
+    private final ExpressionConverter expressionConverter;
+
+    public Accumulator(ExpressionConverter expressionConverter) {
+        this.expressionConverter = expressionConverter;
+    }
+
+    public Optional<String> calculate(String expression) {
+        Optional<String> result = Optional.empty();
+        try {
+            List<String> convert = expressionConverter.convert(expression);
+            BigDecimal calc = getComputationResult(convert).stripTrailingZeros();
+            result = Optional.of(ResultConverter.convertResult(calc.toString()));
+        } catch (IllegalArgumentException | ArithmeticException e) {
+            Console.printError(e.getMessage());
+        }
+        return result;
+    }
+
+    private BigDecimal getComputationResult(List<String> convert) {
+        Deque<BigDecimal> stack = new ArrayDeque<>();
+
+        convert.forEach(op -> {
+            if (Operator.isOperator(op)) {
+                Operator operator = Operator.getOperation(op);
+                BigDecimal n2 = stack.pop();
+                BigDecimal n1 = stack.pop();
+                stack.push(operator.operation(n1, n2));
+            }
+
+            if (!Operator.isOperator(op)) {
+                stack.push(BigDecimal.valueOf(Double.parseDouble(op)));
+            }
+        });
+        return stack.pop();
+    }
```

Comment:
reduce 를 적용하려고 하신 부분이 이곳인가요,,? 

```suggestion
Deque<BigDecimal> result =
    convert.stream().reduce(new ArrayDeque<>(), (stack, op) -> {
    if (Operator.isOperator(op)) {
        BigDecimal n2 = stack.pop();
        BigDecimal n1 = stack.pop();
        Operator operator = Operator.getOperation(op);
        stack.push(operator.operation(n1, n2));
    }
    if(!Operator.isOperator(op)) {
        stack.push(BigDecimal.valueOf(Double.parseDouble(op)));
    }
    return stack;
}, (s1, s2) -> {
    s1.addAll(s2);
    return s1;
});

return result.pop();
```

적용하더라도 큰 차이는 없는 것 같아보입니다..😂

Reply:
헉.. 위와 거의 동일하게 stream을 짰었는데 컴파일 단계에서 계속 오류가 발생해서 사용하지 못했는데... 아마 제가 어디서 뭔가를 잘 못 적은 것 같습니다! 감사합니다!!! 😊